File: console/src/lib/types/protocol/history.ts
import type { MetricSample } from './metrics';

export interface MachineHistory {
  retentionHours: number;
  collectionIntervalSeconds: number;
  nextMetricTimeMs: number | null;
  samples: MetricSample[];
}

export interface HistorySyncResponse {
  serverTimeMs: number;
  machines: Record<string, MachineHistory>;
}

File: hub/src/database/model/history.rs
use gauges_shared::MetricSample;

#[derive(Clone, Debug, PartialEq)]
pub struct StoredHistory {
    pub retention_hours: u64,
    pub collection_interval_seconds: u64,
    pub samples: Vec<MetricSample>,
    pub next_metric_time_ms: Option<i64>,
}
